metrics/query.global/global.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2022 Block, Inc.
+
+package queryglobal
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/cashapp/blip"
+	"github.com/cashapp/blip/metrics/util"
+)
+
+const (
+	DOMAIN = "query.global"
+
+	OPT_RESPONSE_TIME_PERCENTILES = "response-time-percentiles"
+	OPT_RESPONSE_TIME_OPTIONAL    = "response-time-optional"
+	OPT_TRUNCATE_TABLE            = "truncate-table"
+	DEFAULT_PERCENTILE_OPTION     = "999"
+
+	BASE_QUERY     = "SELECT ROUND(bucket_quantile * 100, 1) AS p, ROUND(bucket_timer_high / 1000000, 3) AS us FROM performance_schema.events_statements_histogram_global"
+	TRUNCATE_QUERY = "TRUNCATE TABLE performance_schema.events_statements_histogram_global"
+)
+
+type Global struct {
+	db *sql.DB
+	// --
+	optional      map[string]bool      // keyed on level
+	percentiles   map[string][]float64 // keyed on level
+	truncateTable map[string]bool      // keyed on level
+	available     bool
```

Comment:
`s/available/enabled/`  Perf Schema is always available; question is whether or not it's enabled. (Fwiw, it's enabled by default as of 5.7 iirc, but Aurora _disables_ it by default 🤦 )